### PR TITLE
Improvement/code structure

### PR DESCRIPTION
**Pull request by Taucher2003**
I made some changed that can improve the structure of the code. If you have questions to any of the changes, just comment on the PR and I will most likely answer them.

### CODE REVIEW

File: src/main/java/com/namelessmc/bot/Language.java [hunk 1]
```diff
@@ -180,7 +179,7 @@ private void checkReplacements(final Term term, final Object... replacements) {
 		for (int i = 0; i < replacements.length; i += 2) {
 			Validate.isTrue(replacements[i] instanceof String, "Replacement keys must be strings");
 			final String key = (String) replacements[i];
-			if (key == required[i/2]) {
+			if (Objects.equals(key, required[i / 2])) {
```
Comment (Derkades) [inline src/main/java/com/namelessmc/bot/Language.java:182]: Woops! I don't know how this worked, just lucky with the string pool I guess

File: src/main/java/com/namelessmc/bot/connections/StatelessConnectionManager.java [hunk 2]
```diff
@@ -52,8 +50,8 @@ public boolean removeConnection(final long guildId) throws BackendStorageExcepti
 	}
 
 	@Override
-	public List<URL> listConnections() throws BackendStorageException {
-		return Collections.singletonList(this.api.get().getApiUrl());
+	public List<URL> listConnections() {
+		return this.api.map(namelessAPI -> Collections.singletonList(namelessAPI.getApiUrl())).orElse(Collections.emptyList());
```
Comment (Derkades) [inline src/main/java/com/namelessmc/bot/connections/StatelessConnectionManager.java:54]: Is this needed, isn't the optional always present?

File: src/main/java/com/namelessmc/bot/connections/StatelessConnectionManager.java [hunk 3]
```diff
@@ -73,11 +71,14 @@ public boolean updateConnection(final long guildId, final URL apiUrl) throws Bac
 
 	@Override
 	public Optional<Long> getGuildIdByURL(final URL url) throws BackendStorageException {
-		if (url == this.api.get().getApiUrl()) {
-			return Optional.of(this.guildId);
-		} else {
-			throw new BackendStorageException(new UnsupportedOperationException());
+		if(this.api.isPresent()) {
```
Comment (Derkades) [inline src/main/java/com/namelessmc/bot/connections/StatelessConnectionManager.java:74]: Is this needed, isn't the optional always present?

File: src/main/java/com/namelessmc/bot/http/HttpMain.java [hunk 4]
```diff
@@ -22,18 +21,15 @@ public static void init() {
...
-			@Override
-			public void run() {
-				try {
-					server.start();
-					server.join();
-				} catch (final Exception e) {
-					e.printStackTrace();
-					System.exit(1);
-				}
+		new Thread(() -> {
```
Comment (Derkades) [inline src/main/java/com/namelessmc/bot/http/HttpMain.java:24]: indentation here is not consistent with the rest of the code (tabs)

File: src/main/java/com/namelessmc/bot/listeners/DiscordRoleListener.java [hunk 5]
```diff
@@ -139,10 +138,8 @@ public static void sendRolesToWebsite(final Member member) {
 			user.get().setDiscordRoles(roleIds);
 		} catch (final ApiError e) {
 			Main.getLogger().warning("API error " + e.getError() + " while sending role update for user " + userId + " guild " + guildId + " (setDiscordRoles)");
-			return;
-		} catch (final NamelessException e) {
+        } catch (final NamelessException e) {
```
Comment (Derkades) [inline src/main/java/com/namelessmc/bot/listeners/DiscordRoleListener.java:141]: another indentation issue here

**Review by Derkades — CHANGES_REQUESTED** [review]
Looks good in general, I left some minor comments in a few places. Thanks for contributing!

File: src/main/java/com/namelessmc/bot/connections/StatelessConnectionManager.java (hunk 2 above)
Comment (Taucher2003) [reply src/main/java/com/namelessmc/bot/connections/StatelessConnectionManager.java:54]: should be, but then I don't see the reason for the Optional

File: src/main/java/com/namelessmc/bot/connections/StatelessConnectionManager.java (hunk 3 above)
Comment (Taucher2003) [reply src/main/java/com/namelessmc/bot/connections/StatelessConnectionManager.java:74]: as above, if it is always present, you propably should not use an Optional.

File: src/main/java/com/namelessmc/bot/http/HttpMain.java (hunk 4 above)
Comment (Taucher2003) [reply src/main/java/com/namelessmc/bot/http/HttpMain.java:24]: I have gone through the whole project and made a reformatted the code. It should be consistent over the complete project now.

File: src/main/java/com/namelessmc/bot/listeners/DiscordRoleListener.java (hunk 5 above)
Comment (Taucher2003) [reply src/main/java/com/namelessmc/bot/listeners/DiscordRoleListener.java]: As commented above

File: src/main/java/com/namelessmc/bot/connections/PostgresConnectionManager.java [hunk 6]
```diff
@@ -10,7 +10,7 @@
 	private final int port;
 
 	public PostgresConnectionManager(final String hostname, final int port, final String databaseName,
-			final String username, final String password) {
+									 final String username, final String password) {
```
Comment (Taucher2003) [inline src/main/java/com/namelessmc/bot/connections/PostgresConnectionManager.java:13]: This may look weird in GitHub. In IntelliJ it fits perfect
image

**Review by Derkades — APPROVED** [review]

File: src/main/java/com/namelessmc/bot/connections/StatelessConnectionManager.java (hunk 2 above)
Comment (Derkades) [reply src/main/java/com/namelessmc/bot/connections/StatelessConnectionManager.java:54]: The parent class wants an optional for getGuildIdByURL. For example, in the database storage implementation a guild does not always exist for a URL.

Instead of creating a new Optional.of(url) every time the method is called I chose to create the optional once and return the same optional every time. This has the side effect that listConnections() has an unchecked get() call

Comment (Taucher2003) [reply src/main/java/com/namelessmc/bot/connections/StatelessConnectionManager.java]: I have changed this method a bit. Should look more readable now

File: src/main/java/com/namelessmc/bot/connections/StatelessConnectionManager.java [hunk 7]
```diff
@@ -73,11 +72,14 @@ public boolean updateConnection(final long guildId, final URL apiUrl) throws Bac
 
 	@Override
 	public Optional<Long> getGuildIdByURL(final URL url) throws BackendStorageException {
-		if (url == this.api.get().getApiUrl()) {
-			return Optional.of(this.guildId);
-		} else {
-			throw new BackendStorageException(new UnsupportedOperationException());
+		if (this.api.isPresent()) {
```
Comment (Derkades) [inline src/main/java/com/namelessmc/bot/connections/StatelessConnectionManager.java:75]: Could you revert these changes? It's basically doing this now:
`return optional.isPresent() : Optional.of(optional.get()) : optional.empty()`
which is identical to
`return optional`
